### PR TITLE
Default CHOP namespace to the release namespace

### PR DESCRIPTION
**Pull request by fuziontech**
fixes https://github.com/PostHog/charts-clickhouse/issues/75﻿

### CODE REVIEW

**Comment by macobo** [comment]
Wouldn't this break some existing installations since clickhouseOperator would move?

**Comment by fuziontech** [comment]
> Wouldn't this break some existing installations since clickhouseOperator would move?

Only if they were installing this off label from our instructions would this be a problem. We have them install this in the `posthog` namespace in all of our instructions. Doing this now vs later would only make this transition cleaner and less painful.

> A more sane way is changing the default value to be blank and using `default .Values.clickhouseOperator.namespace .Release.namespace`

I would say _safe_ over _sane_ to be more accurate. The correct syntax would be `{{ .Values.clickhouseOperator.namespace | default .Release.namespace }}`

> In both cases, going strictly by semver this is a breaking change.

This would be breaking, but the remedy would be simply to have existing installations add `.Values.clickhouseOperator.namespace` back to their provided values.yaml file.

File: charts/posthog/values.yaml [hunk 1]
```diff
@@ -21,8 +21,6 @@ sentryDSN:
 clickhouseOperator:
   # -- Whether to install clickhouse. If false, `clickhouse.host` must be set
   enabled: true
-  # -- Which namespace to install clickhouse operator to
-  namespace: "posthog"
```
Comment (tiina303) [inline charts/posthog/values.yaml:25]: keep it just as empty, so we have the description. I believe the default function still works then

**Comment by tiina303** [comment]
lets yes do the defaulting as Karl suggested (with pipes as that's much better way to use functions).

File: charts/posthog/values.yaml (hunk 1 above)
Comment (fuziontech) [reply charts/posthog/values.yaml:25]: perfect

File: charts/posthog/templates/clickhouse_cluster_role.yaml [hunk 2]
```diff
@@ -124,5 +124,5 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: clickhouse-operator
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_cluster_role.yaml:127]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }} 
```

File: charts/posthog/templates/clickhouse_config_map.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: etc-clickhouse-operator-files
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_config_map.yaml:6]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }} 
```

File: charts/posthog/templates/clickhouse_config_map.yaml [hunk 4]
```diff
@@ -134,7 +134,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: etc-clickhouse-operator-confd-files
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_config_map.yaml:137]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/templates/clickhouse_config_map.yaml [hunk 5]
```diff
@@ -143,7 +143,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: etc-clickhouse-operator-configd-files
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_config_map.yaml:146]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/templates/clickhouse_config_map.yaml [hunk 6]
```diff
@@ -193,7 +193,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: etc-clickhouse-operator-templatesd-files
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_config_map.yaml:196]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/templates/clickhouse_config_map.yaml [hunk 7]
```diff
@@ -292,7 +292,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: etc-clickhouse-operator-usersd-files
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_config_map.yaml:295]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/templates/clickhouse_deployment.yaml [hunk 8]
```diff
@@ -3,7 +3,7 @@ kind: Deployment
 apiVersion: apps/v1
 metadata:
   name: clickhouse-operator
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_deployment.yaml:6]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/templates/clickhouse_service.yaml [hunk 9]
```diff
@@ -8,7 +8,7 @@ kind: Service
 apiVersion: v1
 metadata:
   name: clickhouse-operator-metrics
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }}
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_service.yaml:11]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/templates/clickhouse_service_account.yaml [hunk 10]
```diff
@@ -3,5 +3,5 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: clickhouse-operator
-  namespace: {{ .Values.clickhouseOperator.namespace | quote }} 
+  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.namespace }}
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_service_account.yaml:6]:
```suggestion
  namespace: {{ .Values.clickhouseOperator.namespace | default .Release.Namespace }}
```

File: charts/posthog/values.yaml [hunk 11]
```diff
@@ -21,8 +21,8 @@ sentryDSN:
...
+  # -- Which namespace to install clickhouse operator to (defaults to namespace chart is installed to)
+  namespace: ""
```
Comment (tiina303) [inline charts/posthog/values.yaml:25]: If it's empty str, then default function thinks that's a reasonable value only for no value will default be used and that's what we want.

```suggestion
  namespace:
```

**Review by tiina303 — CHANGES_REQUESTED** [review]
I had a feeling the "" didn't work with default, so I tested it, found another typo. Namespace needs to be capitalized after release.

btw simple way to test this:
```
helm template posthog charts/posthog --set cloud="x" -n "blah" | grep "clickhouse_service_account" -A5
```
^ and look at what you see for the namespace

**Pull request merged**